File: code/1006/1006.py
"""
***@Time    : 2019-10-27
***@Author  : hushengquan
***@File    : 1006.py
***@Software: vim
***@type    : pat practice
"""

num = input()
count = len(num)

number = []
for i in range(count):
    number.append(int(num) % 10)
    num = int(num) / 10

if count == 3:
    for i in range(int(number[2])):
        print('B', end='')
    count -= 1
if count == 2:
    for i in range(int(number[1])):
        print('S', end='')
    count -= 1
if count == 1:
    for i in range(int(number[0])):
        print(i + 1, end='')

File: code/1006/1006.cpp
/***********************************************
 > @Time       : 2019-10-29
 > @Author     : hushengquan
 > @File       : 1006.cpp
 > @Software   : vim
 > @type       : pat practice
************************************************/
#include <iostream>
using namespace std;

int main() {
    int num, count = 0;
    cin >> num;

    int value[3] = {0};
    while (num != 0) {
        value[count++] = num % 10;
        num /= 10;
    }

    for (int i = 0; i < value[2]; i++) cout << "B";
    for (int i = 0; i < value[1]; i++) cout << "S";
    for (int i = 0; i < value[0]; i++) cout << i + 1;

    return 0;
}
